data/libs/EquipType.lua

```diff
@@ -325,17 +431,90 @@ function HyperdriveType:OnLeaveHyperspace(ship)
 	end
 end
 
+--==============================================================================
+
 -- NOTE: "sensors" have no general-purpose code associated with the equipment type
-local SensorType = utils.inherits(EquipType, "SensorType")
+---@class Equipment.SensorType : EquipType
+local SensorType = utils.inherits(EquipType, "Equipment.SensorType")
+
+--==============================================================================
 
 -- NOTE: all code related to managing a body scanner is implemented in the ScanManager component
-local BodyScannerType = utils.inherits(SensorType, "BodyScannerType")
+---@class Equipment.BodyScannerType : EquipType
+---@field stats table
+local BodyScannerType = utils.inherits(SensorType, "Equipment.BodyScannerType")
+
+--==============================================================================
+
+---@class Equipment.CabinType : EquipType
+---@field passengers Character[]?
+local CabinType = utils.inherits(EquipType, "Equipment.CabinType")
+
+---@param passenger Character
+function CabinType:AddPassenger(passenger)
+	table.insert(self.passengers, passenger)
+	self.icon_name = "equip_cabin_occupied"
+end
+
+---@param passenger Character
+function CabinType:RemovePassenger(passenger)
+	utils.remove_elem(self.passengers, passenger)
+	self.icon_name = "equip_cabin_empty"
```

Comment:
Q: Some cabins can take more than one passenger; shouldn't the icon only be changed once the cabin is empty?